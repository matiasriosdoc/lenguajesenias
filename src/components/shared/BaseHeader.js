import Colors from '../../res/colors';

export const BaseHeader = {
  title: '',
  headerTintColor: Colors.THEME_SECONDARY,
  headerBackTitle: '',
  headerTitleStyle: {
    fontFamily: 'nunito',
  },
  headerStyle: {
    backgroundColor: Colors.THEME_PRIMARY,
    elevation: 0,
    borderBottomWidth: 1,
    borderBottomColor: Colors.TAB_BAR_ACTIVE_ICON,
  },
};
