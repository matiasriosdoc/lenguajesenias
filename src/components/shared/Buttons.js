import React from 'react';
import { Text, TouchableOpacity, StyleSheet, Dimensions } from 'react-native';

export function CloseButton(props) {
  const { onPress, text } = props;
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    flex: 1,
    width: Dimensions.get('window').width,
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 100,
    backgroundColor: '#01A0C6',
  },
  buttonText: {
    textAlign: 'center',
    fontFamily: 'nunito',
    fontSize: Dimensions.get('window').width / 20,
    color: '#FFFFFF',
  },
});
