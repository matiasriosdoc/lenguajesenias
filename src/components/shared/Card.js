import React from 'react';
import { TouchableOpacity, View, Image, Text, StyleSheet } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Colors from './../../res/colors';
import { getCardWidth, getCardPadding } from './../../util/layoutUtil';

const styles = StyleSheet.create({
  cardContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  imageContainer: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    backgroundColor: 'white',
  },
  image: {
    backgroundColor: Colors.CATEGORY_IMAGE_BACKGROUND_GREY,
  },
  selectable: {
    height: 40,
    width: 40,
    position: 'absolute',
    top: 12,
    right: 12,
    zIndex: 100,
  },
  nameContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'white',
    height: 50,
  },
  name: {
    textAlign: 'center',
    fontFamily: 'nunito',
  },
});

export function Card({ name, children, onPress, src }) {
  const imagePaddingHorizontal = getCardPadding() * 2;
  const imagePaddingVertical = getCardPadding() * 2;

  return (
    <TouchableOpacity
      onPress={onPress}
      style={[
        styles.cardContainer,
        {
          width: getCardWidth(),
          paddingVertical: getCardPadding(),
          paddingHorizontal: getCardPadding(),
        },
      ]}>
      <View
        style={[
          styles.imageContainer,
          {
            width: getCardWidth() - 2 * getCardPadding(),
            height: getCardWidth() - 4 * getCardPadding(),
          },
        ]}>
        {children}
        <Image
          style={[
            styles.image,
            {
              width:
                getCardWidth() -
                2 * (getCardPadding() + imagePaddingHorizontal),
              height:
                getCardWidth() - 2 * (getCardPadding() + imagePaddingVertical),
            },
          ]}
          source={src}
        />
      </View>
      <View
        style={[
          styles.nameContainer,
          {
            width: getCardWidth() - 2 * getCardPadding(),
            paddingHorizontal: imagePaddingHorizontal,
          },
        ]}>
        <Text style={styles.name}>{name}</Text>
      </View>
    </TouchableOpacity>
  );
}

export function SelectableCard({ selected, color, ...props }) {
  return (
    <Card {...props}>
      {selected && (
        <MaterialIcons
          name="check-circle"
          size={40}
          style={styles.selectable}
          color={color}
        />
      )}
    </Card>
  );
}
