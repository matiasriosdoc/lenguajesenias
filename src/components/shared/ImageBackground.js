import React from 'react';
import {
  Dimensions,
  ImageBackground as NatImgBack,
  StyleSheet,
  Platform,
} from 'react-native';

const styles = StyleSheet.create({
  full: {
    flex: 1,
  },
  backgroundImageStyle: {
    resizeMode: Platform.OS === 'ios' ? 'repeat' : 'stretch',
  },
});

export default function ImageBackground(props) {
  return (
    <NatImgBack
      style={props.style || styles.full}
      imageStyle={
        props.style
          ? []
          : [
              styles.backgroundImageStyle,
              {
                width: Dimensions.get('window').width,
                height: Dimensions.get('window').height,
              },
            ]
      }
      resizeMode={props.resizeMode}
      source={props.src}>
      {props.children}
    </NatImgBack>
  );
}
