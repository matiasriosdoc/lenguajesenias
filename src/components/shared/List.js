import React from 'react';
import { FlatList, StyleSheet } from 'react-native';
import { getCardPadding } from './../../util/layoutUtil';

const styles = StyleSheet.create({
  viewContainer: {
    flex: 1,
    paddingVertical: getCardPadding(),
    paddingHorizontal: getCardPadding(),
  },
});

export default function List({ data, scrollRef, renderItem }) {
  return (
    <FlatList
      ref={scrollRef}
      style={styles.viewContainer}
      numColumns={2}
      initialNumToRender={6}
      maxToRenderPerBatch={10}
      removeClippedSubviews={true}
      data={data}
      keyExtractor={item => item.name_es}
      renderItem={renderItem}
    />
  );
}
