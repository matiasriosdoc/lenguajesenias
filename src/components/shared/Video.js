import React, { useState } from 'react';
import {
  TouchableOpacity,
  Image,
  Dimensions,
  StyleSheet,
  View,
} from 'react-native';
import NativeVideo from 'react-native-video';

const margin = 12;
const playIconSize = 100;

const styles = StyleSheet.create({
  videoContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: margin,
    marginBottom: margin,
    backgroundColor: 'white',
  },
  video: {
    //backgroundColor: 'transparent',
  },
  playIcon: {
    height: playIconSize,
    width: playIconSize,
  },
  buttonsContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
    position: 'absolute',
    top: 10,
    left: 10,
  },
  button: {
    marginBottom: 5,
  },
  slowButtonImage:{
    width: 109,
    height: 40,
    resizeMode: 'contain',
  },
  normalButtonImage:{
    width: 91,
    height: 41,
    resizeMode: 'contain',
  },
});

const playIcon = require('../../res/icon/play-icon.png');
const normalVelocityIcon = require('../../res/image/conejo-velocidad-normal.png');
const slowVelocityIcon = require('../../res/image/tortuga-velocidad-lenta.png');
const videoRatio = 352 / 288;

export default function Video({ uri, onEnd, onLoadStart, onReady, autoPlay, style}) {
  const [paused, setPaused] = useState(!autoPlay);
  const videoRef = null;
  const [speed, setSpeed] = useState(1.0);

  const handleSpeedChange = newSpeed => {
    setSpeed(newSpeed);
    videoRef?.current.setNativeProps({ rate: newSpeed });
  };

  const _onEnd = () => {
    onEnd && onEnd();
  };

  const _onLoadStart = () => {
    onLoadStart && onLoadStart();
  };

  const _onReady = () => {
    onReady && onReady();
  };

  const videoWidth = Dimensions.get('window').width - 2 * margin;
  const videoHeight = Math.round(videoWidth / videoRatio);

  return (
    <TouchableOpacity
      style={{...styles.videoContainer,
       marginHorizontal: margin,
       width: videoWidth,
       height: videoHeight,
       ...style,
        }}
      onPress={() => {
        videoRef?.seek(0);
        setPaused(!paused);
      }}>
      {paused ? (
        <Image style={styles.playIcon} source={playIcon} />
      ) : (
        <>
          <NativeVideo
            ref={videoRef}
            source={{ uri }}

            style={{...styles.video,
                width: videoWidth,
                height: videoHeight,
                ...style,
                }}
            rate={speed}
            paused={paused}
            muted
            resizeMode="contain"
            repeat={true}
            onEnd={_onEnd}
            onLoadStart={_onLoadStart}
            onReadyForDisplay={_onReady}
          />
          <View style={styles.buttonsContainer}>
            <TouchableOpacity style={styles.button} onPress={() => handleSpeedChange(0.5)}>
              <Image style={styles.slowButtonImage} source={slowVelocityIcon} />
            </TouchableOpacity>
            <TouchableOpacity style={styles.button} onPress={() => handleSpeedChange(1.0)}>
              <Image style={styles.normalButtonImage} source={normalVelocityIcon} />
            </TouchableOpacity>
          </View>
        </>
      )}
    </TouchableOpacity>
  );
}
