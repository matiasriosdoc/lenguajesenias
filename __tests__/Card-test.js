import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Card } from '../src/components/shared/Card';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('handles on press', () => {
  const mockFn = jest.fn();
  const { getByText } = render(<Card name={'cardName'} onPress={mockFn} />);

  fireEvent.press(getByText('cardName'));

  expect(mockFn).toBeCalled();
});
