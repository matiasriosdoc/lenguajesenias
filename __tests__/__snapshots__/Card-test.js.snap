// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 6,
      "paddingVertical": 6,
      "width": 369,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "justifyContent": "flex-end",
        },
        Object {
          "height": 345,
          "width": 357,
        },
      ]
    }
  >
    <Image
      style={
        Array [
          Object {
            "backgroundColor": "#F2F2F2",
          },
          Object {
            "height": 333,
            "width": 333,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "height": 50,
          "justifyContent": "center",
        },
        Object {
          "paddingHorizontal": 12,
          "width": 357,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "nunito",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
